libraries/src/Installer/InstallerScript.php

```diff
@@ -177,13 +177,13 @@ public function getInstances($isModule)
         $query = $db->getQuery(true);
 
         // Select the item(s) and retrieve the id
-        $query->select($db->quoteName('id'));
-
         if ($isModule) {
-            $query->from($db->quoteName('#__modules'))
+            $query->select($db->quoteName('id'))
+                ->from($db->quoteName('#__modules'))
                 ->where($db->quoteName('module') . ' = :extension');
         } else {
-            $query->from($db->quoteName('#__extensions'))
+            $query->select($db->quoteName('extension_id') . ' AS id')
```

Comment:
```suggestion
            $query->select($db->quoteName('extension_id', 'id'))
```
The column alias (` AS ...`) can also be achieved with the 2nd argument of the `quoteName` method.